### ISSUE TITLE
Codeowner change detection doesn't work for outdated branches

### ISSUE THREAD

**Issue by otbutz** [go-gitea/gitea#28135]
### Description

If the branch of a pull request is behind its target branch, any changes from missing commits on the target branch will also trigger their associated rules.

The codeowner detection should rather be based on the actual diff output that is shown in the pull request page.

### Gitea Version

1.21.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

Binary started via systemd behind a caddy reverse proxy.

### Database

None

**Comment by bt90** [comment]
@cl-bvl I wonder if this intentional behaviour or should the list of files be computed from the actual diff?

**Comment by m3rtl** [comment]
This is also the case for us - is there any news on this?

**Comment by zowers** [comment]
owners of files unrelated to the PR are notified, please consider fixing

**Comment by zowers** [comment]
Caused by how list of changed files is computed: between target branch and pr branch

```go
	changedFiles, err := repo.GetFilesChangedBetween(git.BranchPrefix+pr.BaseBranch, pr.GetGitRefName())
```
here:
https://github.com/go-gitea/gitea/pull/24910/files#diff-bc78d680355495193dadf6796c153e5f1bf55e91c3005e0c3055ee72a7fe2c7cR931 (#24921 )

same in main: https://github.com/go-gitea/gitea/blob/0900c1552b51c5d1d883bd3662e67891a5dac80d/models/issues/pull.go#L925

**Comment by lunny** [comment]
This has been resolved by #29783
